tests/core/fsck-on-boot/task.yaml

```diff
@@ -0,0 +1,86 @@
+summary: the boot base provides essential fsck programs
+# TODO:UC20: this test fails on core16, core18 and core20 and may require
+# separate fixes on each platform. really fails on core20, fsck is not invoked!
+details: |
+  Snapd uses vfat on certain essential boot partitions, due to external
+  requirements imposed by the bootloader architecture. This test verifies that
+  the boot process is capable of detecting unclean fvat and fixing it before
+  such file system is mounted. This is an essential property to ensure
+  longevity of devices that rely on write to vfat to operate.
+prepare: |
+  tests.cleanup prepare
+execute: |
+  unmount_vfat() {
+    case "$SPREAD_SYSTEM" in
+      ubuntu-core-16-*)
```

Comment:
perhaps here you could use the check declared in systems.sh (is_core16_system)

Reply:
I used your new os.query tool instead.